### PR TITLE
ssr & typings

### PR DESCRIPTION
**Pull request by wealth**
actually not sure if works

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -21,16 +21,18 @@ export default class PrintProvider extends React.PureComponent {
...
+			this.setState({ isInPrintPreview: window.matchMedia('print').matches });
+		};
+	}
   }
 
   getChildContext () {
     return {
       printProvider: {
-        isPrint: window.matchMedia('print').matches,
+        isPrint: global.window && window.matchMedia('print').matches,
```
Comment (a-x-) [inline src/index.jsx]: `global.window` is something weird

File: src/index.jsx [hunk 2]
```diff
@@ -21,16 +21,18 @@ export default class PrintProvider extends React.PureComponent {
...
-    window.matchMedia('print').onchange = () => {
-      debug('toggle print mode', window.matchMedia('print').matches);
-      this.setState({ isInPrintPreview: window.matchMedia('print').matches });
-    };
+	if (global.window) {
+		window.matchMedia('print').onchange = () => {
+			debug('toggle print mode', window.matchMedia('print').matches);
+			this.setState({ isInPrintPreview: window.matchMedia('print').matches });
+		};
+	}
```
Comment (a-x-) [inline src/index.jsx]: can you keep 2 space indentation?
:)

**Comment by a-x-** [comment]
closes https://github.com/a-x-/react-easy-print/issues/7

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+declare module "react-easy-print" {
+	export const Print: any;
+	export const NoPrint: any;
+	const PrintProvider: any;
```
Comment (zmilonas) [inline index.d.ts]: I don't think this a proper way to do typing.
It'd be nice for you to actually put the types.

Comment (a-x-) [reply index.d.ts]: +1, @zmilonas,
Can you add correct typings after the pr we’re preparing now: https://github.com/a-x-/react-easy-print/pull/19

Comment (a-x-) [reply index.d.ts]: My reply seems outdated a bit,
I accepted looking correct typings here: https://github.com/a-x-/react-easy-print/pull/12
We’re try to update them after #19 pr will be merged

**Comment by a-x-** [comment]
ts typings are in master branch for now
you can check it:
📦 0.7.1-0
reopen please if it does not work for you

**Pull request closed without merging**